genesyscloud/architect_user_prompt/resource_genesyscloud_architect_user_prompt.go

```diff
@@ -0,0 +1,355 @@
+				if !ok {
+					continue
+				}
+				resourceFilename, ok := resourceMap["filename"].(string)
+				if !ok || resourceFilename == "" {
+					continue
+				}
+				APIResources := *userPrompt.Resources
+				for _, APIResource := range APIResources {
+					if APIResource.Tags == nil {
+						continue
+					}
+					tags := *APIResource.Tags
+					filenameTag, ok := tags["filename"]
+					if !ok {
+						continue
+					}
+					if len(filenameTag) > 0 {
+						if filenameTag[0] == resourceFilename {
+							if *APIResource.UploadStatus != "transcoded" {
+								return retry.RetryableError(fmt.Errorf("prompt file not transcoded. User prompt ID: '%s'. Filename: '%s'", d.Id(), resourceFilename))
+							}
+						}
+					}
+				}
+			}
+		}
+
+		_ = d.Set("resources", flattenPromptResources(d, userPrompt.Resources))
+
+		log.Printf("Read Audio Prompt %s %s", d.Id(), *userPrompt.Id)
+		return cc.CheckState()
+	})
+}
+
+func updateUserPrompt(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	name := d.Get("name").(string)
+	description := d.Get("description").(string)
+
+	sdkConfig := meta.(*provider.ProviderMeta).ClientConfig
+	proxy := getArchitectUserPromptProxy(sdkConfig)
+
+	prompt := platformclientv2.Prompt{
+		Name: &name,
+	}
+
+	if description != "" {
+		prompt.Description = &description
+	}
+
+	log.Printf("Updating user prompt %s", name)
+	_, _, err := proxy.updateArchitectUserPrompt(ctx, d.Id(), prompt)
+	if err != nil {
+		return diag.Errorf("Failed to update user prompt %s: %s", name, err)
+	}
+
+	diagErr := updatePromptResource(ctx, d, proxy, sdkConfig)
+	if diagErr != nil {
+		return diagErr
+	}
+
+	log.Printf("Updated User Prompt %s", d.Id())
+	return readUserPrompt(ctx, d, meta)
+}
+
+func deleteUserPrompt(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	name := d.Get("name").(string)
+
+	sdkConfig := meta.(*provider.ProviderMeta).ClientConfig
+	proxy := getArchitectUserPromptProxy(sdkConfig)
+
+	log.Printf("Deleting user prompt %s", name)
+	if _, err := proxy.deleteArchitectUserPrompt(ctx, d.Id(), true); err != nil {
+		return diag.Errorf("Failed to delete user prompt %s: %s", name, err)
+	}
+	log.Printf("Deleted user prompt %s", name)
+
+	return util.WithRetries(ctx, 30*time.Second, func() *retry.RetryError {
+		_, resp, err, retryable := proxy.getArchitectUserPrompt(ctx, d.Id(), false, false, nil)
+		if retryable {
+			if resp != nil && resp.StatusCode == 404 {
+				// User prompt deleted
+				log.Printf("Deleted user prompt %s", name)
+				return nil
+			}
+			return retry.NonRetryableError(fmt.Errorf("error deleting user prompt %s: %s", name, err))
+		}
+		return retry.RetryableError(fmt.Errorf("user prompt %s still exists", name))
+	})
+}
+
+func updatePromptResource(ctx context.Context, d *schema.ResourceData, proxy *architectUserPromptProxy, sdkConfig *platformclientv2.Configuration) diag.Diagnostics {
```

Comment:
This function and the one below could probably go into the utils file

Reply:
Util file typically shouldn't contain any API calls.